src/op_analytics/datapipeline/chains/mapping_utils.py

```diff
@@ -0,0 +1,372 @@
+from pathlib import Path
+from typing import Any, Dict, List, Optional
+
+import polars as pl
+
+from op_analytics.coreutils.logger import structlog
+
+log = structlog.get_logger()
+
+
+# Mapping rule types supported by the system
+SUPPORTED_MAPPING_TYPES = {
```

Comment:
Tangible use case (if this handles for it):

- We have a chain id collision between shape and molten (360). Does this handle for fixing that (i.e. maybe we overwrite molten's 360 to 360-molten?)